src/webhooks/handleSubscriptionCanceled.ts

```diff
@@ -101,15 +105,20 @@ export default async function handleSubscriptionCanceled(
       log,
     });
   } catch (error) {
+    const err = error as Error;
+    log.error(`[SUB CANCEL/ERROR]: Error canceling tier product. ERROR: ${err.stack ?? err.message}`);
     if (!(error instanceof TierNotFoundError)) {
       throw error;
     }
 
     try {
       await updateUserTier(uuid, FREE_INDIVIDUAL_TIER, config);
     } catch (err) {
-      log.error(`[TIER/SUB_CANCELED] Error while updating user tier: uuid: ${uuid} `);
-      log.error(err);
+      const error = err as Error;
+      log.error(
+        `[SUB CANCEL/ERROR]: Error while updating user tier: uuid: ${uuid}. [ERROR STACK]: ${error.stack ?? error.message} `,
+      );
+      throw err;
```

Comment:
Remove this throw, this is changing the behaviour of the payments service completely. The policy is to change user's conditions first and then the tier things after. (pay attention to the changeStorage call behind)